fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBSystemOperations.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * FDBSystemDatabase.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.Transaction;
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.system.SystemKeyspace;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * View of an FoundationDB database used for accessing system and special keys. These are special keys defined within
+ * special system keyspaces that contain information about the underlying FoundationDB cluster, about the client
+ * configuration, or other information available through another API.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class FDBSystemOperations {
```

Comment:
I agree that these methods have a weird-feeling signature.

How about making `newRunner()` a `FunctionalInterface` implemented by `FDBDatabase` and `FDBRecordContext` and taking one of those?

Reply:
Hmm, yeah, something like that might make sense, though I'm hesitant to add a whole new interface like that as part of this PR. For now, I went with something simpler, namely I removed all of the public methods that take an `FDBRecordContext` and now the only methods that this class takes are `FDBDatabaseRunner`s, which simplifies this a bit, and for these methods, is actually the same efficiency as the "reads" don't require a real database read (and therefore an FDB read version (I don't think). It also seems like something we could add later if I'm wrong about that.